aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -12,64 +12,82 @@ use crate::{
     transaction_metadata::TransactionMetadata,
 };
 use aptos_crypto::HashValue;
+use aptos_gas::{AptosGasParameters, FromOnChainGasSchedule, NativeGasParameters};
 use aptos_logger::prelude::*;
 use aptos_state_view::StateView;
 use aptos_types::{
     account_config::{ChainSpecificAccountInfo, APTOS_CHAIN_INFO, CORE_CODE_ADDRESS},
-    on_chain_config::{ConfigStorage, OnChainConfig, VMConfig, Version, APTOS_VERSION_3},
+    on_chain_config::{GasSchedule, OnChainConfig, Version, APTOS_VERSION_3},
     transaction::{ExecutionStatus, TransactionOutput, TransactionStatus},
     vm_status::{StatusCode, VMStatus},
 };
 use fail::fail_point;
 use move_deps::{
-    move_binary_format::{
-        errors::{Location, VMResult},
-        CompiledModule,
-    },
+    move_binary_format::{errors::VMResult, CompiledModule},
     move_core_types::{
-        account_address::AccountAddress,
-        gas_schedule::{CostTable, GasAlgebra, GasCarrier, GasUnits, InternalGasUnits},
+        gas_schedule::GasAlgebra,
         language_storage::ModuleId,
         move_resource::MoveStructType,
         resolver::ResourceResolver,
         value::{serialize_values, MoveValue},
     },
-    move_vm_runtime::{logging::expect_no_verification_errors, session::Session},
-    move_vm_types::gas_schedule::GasStatus,
+    move_vm_runtime::logging::expect_no_verification_errors,
+    move_vm_types::gas::UnmeteredGasMeter,
 };
 use std::sync::Arc;
 
 #[derive(Clone)]
 /// A wrapper to make VMRuntime standalone and thread safe.
 pub struct AptosVMImpl {
     move_vm: Arc<MoveVmExt>,
-    on_chain_config: Option<VMConfig>,
+    gas_params: Option<AptosGasParameters>,
     version: Option<Version>,
     chain_account_info: Option<ChainSpecificAccountInfo>,
 }
 
 impl AptosVMImpl {
     #[allow(clippy::new_without_default)]
     pub fn new<S: StateView>(state: &S) -> Self {
-        let inner = MoveVmExt::new()
+        let storage = RemoteStorage::new(state);
+
+        // TODO(Gas): this should not panic
```

Comment:
Are you going to address these todos in the follow up PRs?

Reply:
Yes